helpers/executeCommand.js

```diff
@@ -0,0 +1,29 @@
+const shelljs = require('shelljs');
+
+/**
+ * A helper function that execute a single shell command or JavaScript function.
+ * Supports asynchronous JS functions, and executes shell commands in a child process
+ * that the event loop is not blocked while executing the command.
+ * @param  {string | function(): (number | Promise<number>)} command
+ * @return Exit code for the executed command, a non-zero value indicates failure
+ */
+module.exports = async function executeCommands(command) {
+  let code = 0;
+  if (typeof command === 'function') {
+    if (command.name) {
+      console.info(`${command.name}()`);
+    }
+    try {
+      code = await Promise.resolve(command());
+    } catch (e) {
+      console.error(e);
+      code = 1;
+    }
+  } else {
+    const shellCommand = command.replace('\n', '').replace(/\s+/g, ' ').trim();
+    console.info(shellCommand);
+    code = await new Promise(resolve => shelljs.exec(shellCommand, resolve));
```

Comment:
What happens if `shelljs.exec` fails? The promise won't be rejected unless you call `reject`, which you're not doing here?

Reply:
The first argument passed to the callback is the exit code, which would be `1` if the command failed. This function (and also `executeCommands`) does not reject on errors, instead it resolves with a non-zero exit code. Do you think we should rewrite this to reject on non-zero values returned from shell?

Reply:
> Do you think we should rewrite this to reject on non-zero values returned from shell?

Never mind what I said. I misunderstood the design of this. (I'm not sure if rejecting or returning the exit is better.)